file/file.go

```diff
@@ -919,37 +947,37 @@ func (r *Router) GetFileAncestors(c *gin.Context) {
 	permittedAncestors := ancestors[firstPermittedFileIndex:]
 
 	populatedPermittedAncestors := make([]*GetFileByIDResponse, 0, len(permittedAncestors))
+	filesRes, err :=r.fileClient().GetFilesByIDs(c.Request.Context(), &fpb.GetByFilesByIDsRequest{Ids: permittedAncestors})
 
-	for i := 0; i < len(permittedAncestors); i++ {
-		file, err := r.fileClient().GetFileByID(
-			c.Request.Context(),
-			&fpb.GetByFileByIDRequest{Id: permittedAncestors[i]},
-		)
-		if err != nil {
-			httpStatusCode := gwruntime.HTTPStatusFromCode(status.Code(err))
-			loggermiddleware.LogError(r.logger, c.AbortWithError(httpStatusCode, err))
+	if err != nil {
+		httpStatusCode := gwruntime.HTTPStatusFromCode(status.Code(err))
+		loggermiddleware.LogError(r.logger, c.AbortWithError(httpStatusCode, err))
+		return
+	}
+	files := filesRes.GetFiles()
+	fmt.Printf("files: %v\n", files)
 
-			return
-		}
-		res, err := r.favoriteClient().IsFavorite(c, &fvpb.IsFavoriteRequest{UserID: reqUser.ID, FileID: file.GetId()})
+	for file := range files {
```

Comment:
Use the file itself and not the index.
```suggestion
	for index, file := range files {
```